src/js/components/add-image/add-image.js

```diff
@@ -2,12 +2,23 @@
 import React               from 'react';
 import MaharaBaseComponent from '../base.js';
 import Router              from '../../router.js';
-import StateStore, 
+import StateStore,
        {maharaServer}      from '../../state.js';
 import ImageDetails        from './image.js';
 import {FILE_ENTRY, PAGE_URL} from '../../constants.js';
 
 class AddImage extends MaharaBaseComponent {
+  constructor(props) {
+      super(props);
+
+      this.imageToEdit = this.props.pendingUploads.find(upload => upload.guid === this.props.imageToEdit);
+      this.state = {
+          targetFolderName: this.imageToEdit.guid ? this.imageToEdit.targetFolderName : this.props.server.defaultFolderName
+      };
+
+      this.changeFolder = this.changeFolder.bind(this);
+  }
+
     render() {
```

Comment:
I'm guessing this is a larger issue that you should probably fix as a broader refactor of a bunch of things (rather than in this commit), but I think it's a good convention to have the render method last.